pkg/controller/installation/core_controller.go

```diff
@@ -1692,3 +1721,28 @@ func cidrWithinCidr(cidr, pool string) bool {
 	}
 	return false
 }
+
+func addCRDWatches(c controller.Controller, v operator.ProductVariant) error {
+	pred := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			// Create occurs because we've created it, so we can safely ignore it.
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if utils.IgnoreObject(e.ObjectOld) && !utils.IgnoreObject(e.ObjectNew) {
+				// Don't skip the removal of the "ignore" annotation. We want to
+				// reconcile when that happens.
+				return true
+			}
+			// Otherwise, ignore updates to objects when metadata.Generation does not change.
+			return e.ObjectOld.GetGeneration() != e.ObjectNew.GetGeneration()
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool { return true },
+	}
+	for _, x := range crds.GetCRDs(v) {
```

Comment:
Maybe this doesn't matter, but I wonder if we should not return the full CRD definitions just the get the names. This leaves a potentially large, albeit it very temporary, memory footprint since we're storing the full CRD definitions in code. Having a subsequent `GetCRDNames` function that iterates through the CRDs and pulls out the names while only ever storing one CRD at a time might be a little less costly.

Reply:
You could even have a custom struct instead of the full `&apiextenv1.CustomResourceDefinition{}` resource that only has room for the meta data so that not even the full CRD is parsed (which will alleviate initial CPU usage).

Reply:
Right, I didn't realise you were storing that globally. How much space does it take to keep those CRDs in memory?

I'm wondering if we should include that CRD watch in the installation controller vs having the installation controller write the CRDs once on startup then have a CRD controller watch for the individual CRD changes. That way if a CRD changes we could just have a map of CRD names to file (created at startup) and load the specific CRD that we detected was changed. This would give us improved CPU and memory, but it complicates the code base a little more.

Reply:
Laurence is actually working on changes that add in a dedup of our Reconcile since we fully reconcile on each pass. So I think what you are suggesting would be counter productive to that change.

I think I'd be opposed to optimizing this more as I think my time spent optimizing with those globals was probably unwarranted since I didn't have any evidence that it was needed.